src/python/module/z5py/file.py

```diff
@@ -48,6 +49,8 @@ def infer_format(cls, path):
 
     def __init__(self, path, mode='a', use_zarr_format=None):
 
+        if isinstance(path, pathlib.Path):
+            path = str(path)
```

Comment:
I don't have much experiences with pathlib, but it looks like @clbarnes suggestion is a bit more general.
Is there a reason against doing it that way?

Reply:
> I think the maximally correct way of doing this would be to check `isinstance(path, os.PathLike)`, and use `os.fspath()` instead of `str` to convert. Not seen many `PathLike`s outside of `pathlib` though.

Hey @clbarnes, thanks a lot for looking at it. i didn't know about `os.PathLike`. Will of course use that instead :)

Reply:
I think `os.PathLike` basically just guarantees an `__fspath__` method, so in true python fashion, just calling `os.fspath` and catching the resulting exception is probably just as good.